src/content-feature.js

```diff
@@ -87,14 +90,32 @@ export default class ContentFeature {
      */
     get messagingContext () {
         if (this.#messagingContext) return this.#messagingContext
+
+        const contextName = import.meta.injectName === 'apple-isolated'
```

Comment:
We could just land the messaging refactoring + extension support as an isolated PR and discuss the other parts after.